src/overlays/actors/ovl_En_Onpuman/z_en_onpuman.c

```diff
@@ -27,29 +28,150 @@ const ActorInit En_Onpuman_InitVars = {
+        AT_NONE,
+        AC_ON | AC_TYPE_PLAYER,
+        OC1_ON | OC1_TYPE_ALL,
+        OC2_TYPE_1,
+        COLSHAPE_CYLINDER,
+    },
+    {
+        ELEMTYPE_UNK0,
+        { 0x00000000, 0x00, 0x00 },
+        { 0x00000000, 0x00, 0x00 },
+        TOUCH_NONE | TOUCH_SFX_NORMAL,
+        BUMP_ON,
+        OCELEM_ON,
+    },
     { 30, 40, 0, { 0, 0, 0 } },
 };
 
-#endif
-
-extern ColliderCylinderInit D_80B12390;
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Onpuman/EnOnpuman_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Onpuman/EnOnpuman_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Onpuman/func_80B11F44.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Onpuman/func_80B11F78.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Onpuman/func_80B1202C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Onpuman/func_80B1217C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Onpuman/func_80B121D8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Onpuman/EnOnpuman_Update.s")
+void EnOnpuman_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnOnpuman* this = THIS;
+
+    ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawCircle, 36.0f);
+    this->actor.flags |= 0x02000000;
+    Collider_InitAndSetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+    this->actor.colChkInfo.mass = 0xFF;
+    Actor_SetScale(&this->actor, 0.01f);
+    this->actor.targetMode = 6;
+    this->unk_2A4 = 0;
+    this->unk_2A0 = NULL;
+    this->actionFunc = func_80B121D8;
+    this->actor.velocity.y = -10.0f;
+    this->actor.terminalVelocity = -10.0f;
+    this->actor.gravity = -5.0f;
+}
+
+void EnOnpuman_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnOnpuman* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+s32 func_80B11F44(GlobalContext* globalCtx) {
+    Actor* itemAction = globalCtx->actorCtx.actorLists[ACTORCAT_ITEMACTION].first;
+
+    if (itemAction != NULL) {
+        if (itemAction->id != ACTOR_EN_GAKUFU) {
+            itemAction = itemAction->next;
+        }
+        return itemAction;
+    }
+    return 0;
+}
+
+void func_80B11F78(EnOnpuman* this, GlobalContext* globalCtx) {
+    if (globalCtx->msgCtx.ocarinaMode == 4) {
+        this->actionFunc = func_80B121D8;
+        if (this->actor.cutscene != -1) {
+            ActorCutscene_Stop(this->actor.cutscene);
+        }
+    } else if (globalCtx->msgCtx.ocarinaMode == 3) {
+        play_sound(NA_SE_SY_CORRECT_CHIME);
+        globalCtx->msgCtx.ocarinaMode = 4;
+        if (this->actor.cutscene != -1) {
+            ActorCutscene_Stop(this->actor.cutscene);
+        }
+        this->actionFunc = func_80B121D8;
+    }
+}
+
+void func_80B1202C(EnOnpuman* this, GlobalContext* globalCtx2) {
+    GlobalContext* globalCtx = globalCtx2;
+
+    if ((Message_GetState(&globalCtx->msgCtx) == 5) && (Message_ShouldAdvance(globalCtx) != 0)) {
```

Comment:
This returns an `== 0` check (see z_message), so doesn't need another one
```suggestion
    if ((Message_GetState(&globalCtx->msgCtx) == 5) && Message_ShouldAdvance(globalCtx)) {
```
It's a good idea to look elsewhere for patterns like this to see how they've been done before.